### PR TITLE
Initial database population with bloggers and posts data

### LINKED ISSUE #88: First launch of application should populate blogs and blog posts data
When application is launched for the first time and the database is empty, JVM Bloggers should automatically fetch blogs list from JSON file and populate tables with initial set of records.

This will be especially useful if we reduce frequency of schedulers, user would be able to see real data right away after running project locally.

### PR DESCRIPTION
**Pull request by goostleek**
Fixes #88

### CODE REVIEW

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            1 * blogRepository.count() >> 1
+            1 * blogPostRepository.count() >> 1
+        when:
+            tested.initializeDatabaseWithBlogDataIfEmpty()
+        then:
+            0 * bloggersDataFetchingScheduler.fetchBloggersData();
+            0 * blogPostsFetchingScheduler.checkRssForNewBlogPosts();
+    }
+
+    }
```
Comment (cslysy) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy]: Please correct indentation

File: src/main/java/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTrigger.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.BlogPostsFetchingScheduler;
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPostRepository;
+import pl.tomaszdziurko.jvm_bloggers.blogs.BloggersDataFetchingScheduler;
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogRepository;
+
+/**
+ * Initializes database with bloggers and posts data if database is empty.
+ * Usually for the first time application starts.
```
Comment (tdziurko) [inline src/main/java/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTrigger.java]: Add `author` :)

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.BlogPostsFetchingScheduler;
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPostRepository
+import pl.tomaszdziurko.jvm_bloggers.blogs.BloggersDataFetchingScheduler;
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogRepository;
+import spock.lang.Specification;
+import spock.lang.Subject
+
+class InitialBlogDataPopulationTriggerSpec extends Specification {
+
+    def BlogRepository blogRepository = Mock()
```
Comment (tdziurko) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy]: Either `def` or `BlogRepository`

I like this approach:

    BlogRepository blogRepository = Mock(BlogRepository)

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    def BlogPostRepository blogPostRepository = Mock()
+    def BloggersDataFetchingScheduler bloggersDataFetchingScheduler = Mock();
+    def BlogPostsFetchingScheduler blogPostsFetchingScheduler = Mock();
+
+    @Subject
+    def InitialBlogDataPopulationTrigger tested = new InitialBlogDataPopulationTrigger(blogRepository, blogPostRepository, bloggersDataFetchingScheduler, blogPostsFetchingScheduler)
+
+    def "Should trigger bloggers data population if there is no data"() {
+        given:
+            1 * blogRepository.count() >> 0
```
Comment (tdziurko) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy]: Do we need this `1 *` in _given_ section?

Comment (goostleek) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy]: It's a verification if the method was actually invoked (like 'verify(blogrepository).count()' in Mockito). It ensures the test will fail if the method call was removed from 'InitialBlogDataPopulationTrigger'.

File: src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy (hunk 3 above)
Comment (goostleek) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy]: Class name is redundant in this case. Mock type will be inferred from the variable type, so it should be either `BlogRepository blogRepository = Mock()` or `def blogRepository = Mock(BlogRepository)`. Please decide.

Comment (tdziurko) [inline src/test/groovy/pl/tomaszdziurko/jvm_bloggers/InitialBlogDataPopulationTriggerSpec.groovy]: Can be this way.

**Comment by tdziurko** [comment]
We totally forgot about updating Readme (section about running locally) :)

**Pull request merged**